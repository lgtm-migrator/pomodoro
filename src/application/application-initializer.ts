import { IAuthenticationController } from "./authentication-controller";
import { IAuthenticationPresenter } from "./authentication-presenter";
import { IInfrastructureInitializer } from "./infrastructure-initializer";

export class ApplicationInitializer {
  constructor(
    private readonly infrastructureInitializer: IInfrastructureInitializer,
    private readonly authenticationController: IAuthenticationController,
    private readonly authenticationPresenter: IAuthenticationPresenter
  ) {}

  public async initialize(): Promise<void> {
    await this.infrastructureInitializer.initialize();
    this.authenticationPresenter.presentSignedIn(
      await this.authenticationController.isSignedIn()
    );
  }
}
