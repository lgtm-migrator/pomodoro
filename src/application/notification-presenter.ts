export interface INotificationPresenter {
  presentNotification(message: string): void;
}
