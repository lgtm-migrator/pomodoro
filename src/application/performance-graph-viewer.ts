import { Duration, DateTime } from "luxon";
import { DateSerializer } from "../domain/date-serializer";
import { IPerformanceRecordRepository } from "./performance-record-repository";
import { IPerformanceGraphPresenter } from "./performance-graph-presenter";

export class PerformanceGraphViewer {
  constructor(
    private readonly performanceRecordRepository: IPerformanceRecordRepository,
    private readonly performanceGraphPresenter: IPerformanceGraphPresenter
  ) {}

  public async viewGraph(): Promise<void> {
    const records = await this.performanceRecordRepository.findManySince(
      DateSerializer.serialize(
        DateTime.fromJSDate(new Date())
          .minus(Duration.fromObject({ months: 1 }))
          .toJSDate()
      )
    );

    this.performanceGraphPresenter.presentGraph({
      points: records.map(({ date, seconds }) => ({
        date,
        pomodoros: seconds / 25 / 60
      }))
    });
  }
}
