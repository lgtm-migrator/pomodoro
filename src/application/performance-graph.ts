export interface IPerformanceGraph {
  points: { date: string; pomodoros: number }[];
}
