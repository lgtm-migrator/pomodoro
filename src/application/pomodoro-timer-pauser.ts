import { PomodoroTimer } from "./pomodoro-timer";

export class PomodoroTimerPauser {
  constructor(private readonly pomodoroTimer: PomodoroTimer) {}

  public pause(): void {
    this.pomodoroTimer.pause();
  }
}
