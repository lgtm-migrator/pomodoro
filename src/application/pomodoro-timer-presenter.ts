import { PomodoroTimerState } from "./pomodoro-timer-state";
import { ITimerPresenter } from "./timer-presenter";

export interface IPomodoroTimerPresenter extends ITimerPresenter {
  presentTime(seconds: number): void;
  presentState(state: PomodoroTimerState): void;
}
