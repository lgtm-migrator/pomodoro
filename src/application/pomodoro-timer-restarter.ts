import { PomodoroTimer } from "./pomodoro-timer";

export class PomodoroTimerRestarter {
  constructor(private readonly pomodoroTimer: PomodoroTimer) {}

  public restart(): void {
    this.pomodoroTimer.restart();
  }
}
