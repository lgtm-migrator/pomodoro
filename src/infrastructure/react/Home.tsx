import React from "react";
import styled from "styled-components";
import { SignOut } from "./SignOut";
import { Timer } from "./Timer";

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

const SignOutContainer = styled.div`
  position: fixed;
  top: 0.5em;
  right: 0.5em;
`;

export interface IProps {
  signOut: () => Promise<void>;
}

export const Home = ({ signOut }: IProps) => {
  return (
    <Container>
      <Timer />
      <SignOutContainer>
        <SignOut signOut={signOut} />
      </SignOutContainer>
    </Container>
  );
};
