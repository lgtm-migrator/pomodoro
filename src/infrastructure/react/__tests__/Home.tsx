import { create } from "react-test-renderer";
import React from "react";
import { Home } from "../Home";
import { PomodoroTimerState } from "../../../application/pomodoro-timer-state";

it("renders", () => {
  expect(
    create(
      <Home
        performanceGraph={{ data: [] }}
        seconds={42}
        signOut={async () => {}}
        startTimer={async () => {}}
        state={PomodoroTimerState.Pomodoro}
        stopTimer={async () => {}}
        stopped={false}
        viewGraph={async () => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
