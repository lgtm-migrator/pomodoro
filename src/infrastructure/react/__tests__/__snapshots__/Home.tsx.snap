// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-iwsKbI jjzenC"
>
  <div
    className="sc-bxivhb fnePsM"
  >
    <div
      className="sc-EHOje iAwqOV"
    >
      <span
        className="sc-bZQynM bOrPFB"
      >
        0
      </span>
      <span
        className="sc-gzVnrw hPcPGm"
      >
        42
      </span>
    </div>
    <button
      className="sc-bdVaJa sc-htpNat bOjvbW"
      onClick={[Function]}
    >
      Stop
    </button>
  </div>
  <div
    className="sc-gZMcBi gupvCx"
  >
    <button
      className="sc-bdVaJa sc-bwzfXH davee"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </button>
    <button
      className="sc-bdVaJa sc-bwzfXH davee"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 64h64v384h-64zM80 336h64v112H80zM272 272h64v176h-64zM368 176h64v272h-64z"
        />
      </svg>
    </button>
  </div>
</div>
`;
